src/appshell/view/firstlaunchsetup/themespagemodel.cpp

```diff
@@ -96,17 +119,18 @@ QString ThemesPageModel::currentThemeCode() const
 
 void ThemesPageModel::setCurrentThemeCode(const QString& themeCode)
 {
-    if (themeCode == currentThemeCode()) {
+    if (themeCode == currentThemeCode() && !isFollowSystemTheme()) {
         return;
     }
 
+    setFollowSystemTheme(false);
```

Comment:
This logic is duplicated in two places. It should be rather at the configuration level (so that you can call
`uiConfiguration()->setCurrentTheme(...)` from any part of the program without additionally calling `setFollowSystemTheme(false)`)

Reply:
Thanks, done!